app/controllers/admin/deals_controller.rb

```diff
@@ -54,6 +54,19 @@ def destroy
       end
     end
 
+    def publish
+      if @deal.update(published_at: Time.current)
+        render json: { id: @deal.id, published_at: @deal.published_at.to_s(:long) }
+      else
+        render json: { id: @deal.id, status: 'error', error_message: @deal.errors.full_messages }
+      end
+    end
+
+    def unpublish
+      @deal.update(published_at: nil)
+      render json: { id: @deal.id }
```

Comment:
what if we got any errors.
Please always do handle error case, even if it there isn't one for now, it can be there in future and no one will remember to check controller code again.

Always handle failure cases where you should

Reply:
Make sure not to repeat this one